src/NBitcoin/IBitcoinSerializable.cs

```diff
@@ -69,6 +69,9 @@ public static string ToHex(this IBitcoinSerializable serializable, Serialization
         public static void FromBytes(this IBitcoinSerializable serializable, byte[] bytes, ProtocolVersion version = ProtocolVersion.PROTOCOL_VERSION, 
             NetworkOptions options = null)
         {
+            if (ReferenceEquals(null, options) && serializable is IHaveNetworkOptions haveNetworkOptions)
```

Comment:
is this better than `options == null`?

Reply:
Yes. A simple equality test may fail under certain "casting" scenarios affecting null due to the type its being compared to. In this case we need to know its null and not something equivalent to null.